### DISCUSSION TITLE
Running Simulator on Mac Apple silicon

### DISCUSSION THREAD

**Discussion by dzungpv** [revk/xESP32-Faikout#489] [General]
I can build the simulator on the Mac OS M2 apple chip, but it not working.
The module not read or write anything to simulator.
I use SerialTest app https://github.com/wh201906/SerialTest and read/write from the module port. Seem the simulator broken on the Mac.
I run with command: `./faikin-s21 -p /dev/tty.wchusbserial110 -s FTXF20D5V1B.settings -v`

**Comment by dzungpv** [comment]
I try on windows it is the same. @Sonic-Amiga how to run the Simulator? I do not have the adapter but I flash the Faikin and run the command without working. On windows it is: 
`faikin-s21 -p /dev/ttyS0 -s FTXF20D5V1B.settings -v`

**Comment by Sonic-Amiga** [comment]
Your commands are correct. You can in fact run without -s; defaults are sane. Except there's no /dev/something on Windows. Or do you mean Cygwin?
In fact i have never verified it on Cygwin. I just believe it works because UNIX specifics is based on original RevK's code. I will test on systems i have.

I am routinely running the sim on plain Windows (mingw build) with no problems at all.

⁣ King regards 
​

**Reply by dzungpv** [reply]
Yes, Cygwin on Windows and MacOS native build on the Apple Silicon never work.

**Comment by Sonic-Amiga** [comment]
By the way, to eliminate stupid option, did you criss-cross tx/rx in connecting Faikin to PC? Faikin rx should go to pc rx and vice versa.


⁣ King regards 
​

**Comment by Sonic-Amiga** [comment]
No stupid options, the problem confirmed and reproduced on Cygwin. Working on it.
You didn't mention that the code was quite broken. Nevermind, i fixed the build in my repo.

**Reply by dzungpv** [reply]
Yes, If fix some error and build, I want to create PR if it work, but you have done it. I will try your fix on the Cygwin

**Comment by Sonic-Amiga** [comment]
https://github.com/revk/ESP32-Faikin/pull/492
Or my master branch. That one also includes protocol v3 support; but a but not complete, you can't downgrade to v2 or v1. On my TODO list; but no time today, sorry. Falling asleep.

**Reply by dzungpv** [reply]
@Sonic-Amiga 
Windows confirm work with Mingw not Cygwin. But not with MacOS. I try both Intel Mac and Apple Silicon. It can build but not working.
The Cygwin I use the latest GCC, it show this error:
```
C:\dev\esp32\ESP8266-Faikin\Tools\Simulators>faikin-s21 -p COM8
Warning: '/dev/shm' does not exists or is not a directory.

POSIX shared memory objects require the existance of this directory.
Create the directory '/dev/shm' and set the permissions to 01777.
For instance on the command line: mkdir -m 01777 /dev/shm
Could not open shared memory object: Invalid argument
Failed to create shared memory
```
The dev/shm does exist. I revert to previous commit it can run but not working.
I edit the makefile and can build faikin-s21 with MingW but not faikin-x50 and faikin-s21 does work without you last [commit](https://github.com/Sonic-Amiga/ESP8266-Faikin/commit/a354652a6c33e7739f89a69ff99006ff5fdbeae4), the latest code I still not test.
You say you test Cygwin? What is your systems. Mine is Windows 11 x64  

@revk you can run simulator on your Mac?

**Reply by dzungpv** [reply]
On the MacOSX I fix the issue in #493

**Comment by Sonic-Amiga** [comment]
Thank you for cooperation and for sharing your experience. I don't have Mac myself. Didn't know they still foster callout units; modems don't exist for about 30 years...
